### PR TITLE
feat: scheduled tasks (tick) + credential management (env) + notify + AGENTS.md

### PR DESCRIPTION
**Pull request by andrewagrahamhodges**
The agentic-cron primitive: the agent (via Telegram) sets up recurring background work for itself; tasks run silently by default and the agent calls \`phantombot notify\` only when something material happens. Plus the supporting infrastructure: credential discipline, README clarifications, and a new repo-level AGENTS.md.

## What you can do after this merges

**Schedule work**: tell Phantom on Telegram *"every hour, check my email and let me know if anything important comes in."* Phantom runs:

\`\`\`bash
phantombot task add \\
  --schedule \"0 * * * *\" \\
  --description \"hourly email check\" \\
  --prompt \"Check my Gmail since the last run. If anything is important, call \\\`phantombot notify --message \\\"…\\\"\\\`. Reply NONE otherwise.\"
\`\`\`

**Manage tasks** anywhere: *"list my scheduled tasks"* / *"cancel the email check"* — the agent runs \`phantombot task list\` / \`phantombot task cancel <id>\` via Bash.

**Get notified** only when needed: \`phantombot notify --message \"...\"\` (text) or \`--voice \"...\"\` (TTS via the existing voice provider, sent as a Telegram voice note). Silence is the default.

**Manage credentials**: when you give Phantom a new \`GITHUB_TOKEN\`, it runs \`phantombot env set GITHUB_TOKEN \"value\"\` (atomic write, mode 0o600), acks by name only, and the next \`phantombot tick\` already has it in process.env.

## Decisions baked in (from the design conversation)

- **Prompt-only actions** — no \`script:\` action type. The agent reasons via the harness on each fire.
- **Skip missed runs** — \`recordRun\` advances \`next_run_at\` strictly AFTER \`now\`, not after the previous slot. Box off for 5 hours → no avalanche of catch-up runs.
- **Notification opt-in** — tasks are silent by default; the agent calls \`phantombot notify\` if it judges the user needs to know.
- **Self-review prevents task accretion** — every task has \`next_review_at\` scaled to cadence (hourly→14d, daily→30d, weekly→90d). Review prompts ask KEEP / STOP / MODIFY; KEEP doubles the interval (review fatigue is the failure mode); STOP deactivates + notifies.
- **Two .env files, two roles** — \`~/.config/phantombot/.env\` (phantombot-managed, TTS keys) and \`~/.env\` (user-managed, agent writes via \`phantombot env set\`). Both sourced into systemd \`EnvironmentFile=\`.
- **Credential discovery + hygiene** baked into every persona's system prompt (\`CREDENTIALS_SECTION\` in \`src/persona/builder.ts\`).
- **Tick lockfile** at \`\$XDG_STATE_HOME/phantombot/tick.lock\` prevents overlapping ticks.

## New CLI surface

\`\`\`
phantombot task add --schedule \"<cron>\" --prompt \"...\" --description \"...\"
phantombot task list / show <id> / cancel <id>
phantombot tick                          # called by phantombot-tick.timer

phantombot notify --message \"...\"
phantombot notify --voice   \"...\"

phantombot env set NAME \"value\"
phantombot env get NAME
phantombot env list                      # names only — never values
phantombot env unset NAME
\`\`\`

## New systemd unit

\`phantombot-tick.{service,timer}\`. 1-minute cadence (\`AccuracySec=1s\`). Plumbed through \`InstallOptions\` / \`RunInstallInput\` as path overrides matching the #44 isolation pattern. \`uninstall\` stops/disables it first.

All four units (\`main\`, \`heartbeat\`, \`nightly\`, \`tick\`) now source \`~/.config/phantombot/.env\` AND \`~/.env\` via dual \`EnvironmentFile=\` lines. \`ensureUnitCurrent\` (from PR #33) detects the unit-body change as stale on already-installed boxes and rewrites them automatically the next time \`phantombot voice\` / \`phantombot harness\` / etc. runs. Self-upgrading.

## Persona model surprise — clarified in README + AGENTS

The README's new **Personas** section explicitly states that **only one persona is active at runtime** (the surprise from this conversation). Multiple persona directories on disk = library; runtime is single-bound via \`config.defaultPersona\` + \`runLock\`. Switching is config-edit + restart; per-persona memory survives the switch.

## AGENTS.md (new, root-level)

Initialized at \`/AGENTS.md\` for any agent contributing to this codebase (human or LLM). Distinguished from \`personas/<name>/AGENTS.md\` (persona file). Contains the contributing discipline (**\"every PR that changes user-facing behavior must update README + AGENTS in the same PR\"**), build/test/typecheck commands, repo layout, architecture overview, persona model, systemd model, credential model, release pipeline, how-to-add-a-subcommand checklist, testing conventions, common pitfalls (incl. baseline-x64-mandatory, sed-not-printf for version injection, sudo-cp-leaves-root-owned-bak from #47).

## Files

**New (10 source + tests + docs)**:
- \`src/cli/{env,notify,task,tick}.ts\`
- \`src/lib/{cronSchedule,tasks}.ts\`
- \`tests/{cli-env,cli-notify,cli-task,cli-tick,lib-cronSchedule,lib-tasks,persona-builder-credentials}.test.ts\`
- \`AGENTS.md\` (root level — repo contributor guide)

**Modified**:
- \`src/cli/index.ts\` — register env / notify / task / tick subcommands
- \`src/cli/install.ts\` — pass tick paths through
- \`src/lib/systemd.ts\` — tick unit generators + dual EnvironmentFile=
- \`src/persona/builder.ts\` — adds CREDENTIALS_SECTION to every system prompt
- \`src/config.ts\` — adds \`xdgStateHome()\` for the tick lockfile path
- \`tests/cli.test.ts\`, \`tests/cli-install.test.ts\`, \`tests/lib-systemd.test.ts\` — fixtures updated for the new units
- \`README.md\` — major rewrite with Personas, Scheduled Tasks, Notify, Credentials sections + the four-unit systemd model
- \`package.json\`, \`bun.lock\` — adds \`cron-parser\` dep

## Test plan

- [x] \`bun tsc --noEmit\` clean
- [x] \`bun test\` — 434 pass / 1 skip / 0 fail (was 370; +64 new tests)
- [x] Local \`bun run build\` produces a binary; \`./dist/phantombot --help\` shows the new subcommands
- [ ] **Self-test on merge**: workflow run produces \`v1.0.<this PR number>\` with both binaries
- [ ] **Real-world**: after merge, on kw-openclaw run \`phantombot update --force --restart\` → land on the new release; \`phantombot install\` to write the new tick units (or wait for ensureUnitCurrent to upgrade automatically on next \`phantombot voice\`); test \`phantombot task add\` from the agent via Telegram.

### CODE REVIEW

**Review by andrewagrahamhodges — COMMENTED** [review]
## Review: ✅ Approve (one must-fix, a few nits)

Solid work. 64 new tests, good architecture decisions. One thing must change before merge — the rest are suggestions.

### 🔴 Must fix

**`CREDENTIALS_SECTION` uses "kai's"** — this gets injected into **every** persona (robbie, jake, future ones). Should say `phantombot's` or `the agent's`. Flagged inline.

### 🟡 Worth fixing (non-blocking)

Several other files reference Kai by name in generic infrastructure code — `env.ts`, `systemd.ts`, `notify.ts`, `telegram.ts`, `voice.ts`, `binaryUpdate.ts`. These work fine but make the codebase read like a personal project rather than a tool anyone can adopt. Flagged inline on the worst offenders.

### ✅ What's good
- Skip missed runs (no avalanche)
- Silent-by-default tasks
- Self-review KEEP/STOP
- Persona-per-task correct
- Dual EnvironmentFile= clean
- Lockfile + atomic writes
- Tests cover edge cases well

File: src/persona/builder.ts [hunk 1]
```diff
@@ -102,3 +108,75 @@ when KB notes get created or updated based on what you captured during
...
+
+## Discovery — where to find them when a task needs them
+
+Look in this order; don't ask the user for anything that's already
+discoverable:
+
+  1. process.env  — already loaded; phantombot sources both \`~/.env\` and
+                    \`~/.config/phantombot/.env\` via systemd EnvironmentFile=,
+                    so most credentials are available without re-reading.
+  2. ~/.env       — kai's general credentials file. The canonical home for
```
Comment (andrewagrahamhodges) [inline src/persona/builder.ts]: 🔴 Must fix: This CREDENTIALS_SECTION gets injected into every persona's AGENTS.md (robbie, jake, future ones). "kai's" should be "phantombot's" or "the agent's".

File: src/cli/env.ts [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+ * This is the agent's sanctioned write path. Direct `echo … >> ~/.env`
+ * by the harnessed agent would lose atomic-rename + 0o600 — the same
+ * race PR #31 fixed for phantombot's own .env. Going through this
+ * subcommand reuses src/lib/envFile.ts so the agent inherits those
+ * guarantees for free.
+ *
+ * What lives where:
+ *   ~/.config/phantombot/.env  — phantombot's own runtime secrets
+ *                                 (TTS keys; written by `phantombot voice`).
+ *   ~/.env                     — kai's general-purpose credentials
```
Comment (andrewagrahamhodges) [inline src/cli/env.ts]: 🟡 Same "kai's" pattern — this is generic env tooling, not Kai-specific. Suggest "the agent's".

File: src/lib/systemd.ts [hunk 3]
```diff
@@ -16,6 +16,25 @@ export const HEARTBEAT_SERVICE_NAME = "phantombot-heartbeat.service";
...
 export const NIGHTLY_SERVICE_NAME = "phantombot-nightly.service";
 export const NIGHTLY_TIMER_NAME = "phantombot-nightly.timer";
+export const TICK_SERVICE_NAME = "phantombot-tick.service";
+export const TICK_TIMER_NAME = "phantombot-tick.timer";
+
+/**
+ * Both .env files we source into every phantombot unit:
+ *   ~/.config/phantombot/.env  — phantombot's own runtime secrets
+ *                                 (TTS keys; written by `phantombot voice`).
+ *   ~/.env                     — kai's general-purpose credentials
```
Comment (andrewagrahamhodges) [inline src/lib/systemd.ts]: 🟡 Duplicate of the env.ts comment — "kai's" in generic systemd module. Suggest "the agent's".

File: src/cli/notify.ts [hunk 4]
```diff
@@ -0,0 +1,160 @@
+/**
+ * `phantombot notify` — agent's tool for talking to the user out-of-band.
+ *
+ * Why this exists: by design, scheduled tasks (`phantombot tick`) don't
+ * automatically notify Telegram on every run — the user explicitly asked
+ * for silence as default. The harnessed agent calls `phantombot notify`
+ * inside its prompt when it decides the user should hear about something.
+ *
+ * Sends to every chat in `channels.telegram.allowed_user_ids`. For the
+ * common single-allowlisted-user case (kai's setup) that's exactly one
```
Comment (andrewagrahamhodges) [inline src/cli/notify.ts]: 🟡 "kai's setup" as example — reads as personal project. Could say "single-user deployment" or "the default case".
